### DISCUSSION TITLE
Execute code in getServerSideProps only once at first visit?

### DISCUSSION THREAD

**Discussion by schriker** [vercel/next.js#15130] [Help] [answered]
Is there a way to fire some part of code in getServerSideProps only once at first user visit?

I have an app which makes a request to external API for a data which will be used in global scope of app in _app file, so no matter which page user visit that data gets fetched only once on server side:

[https://github.com/schriker/jar-next/blob/master/pages/_app.tsx#L15](https://github.com/schriker/jar-next/blob/master/pages/_app.tsx#L15)

If I use getInitialProps in pages - that works, the problem starts when I want to migrate to getServerSideProps that method fires only on server, so my API call gets triggered every time user navigate to a page with getServerSideProps method which in my case is unnecessary. Is there a way to somehow, execute thath code only once?

**Comment by arunoda** [comment, marked as answer]
> Is there a way to fire some part of code in getServerSideProps only once at first user visit?

In the first time, you can run that logic and add a cookie.
Then in the next time, you can check cookie and ignore that logic.

May be this helps.
